blowfish2.go

```diff
@@ -18,80 +19,72 @@ const (
 )
 
 func NewBlowfish2(key, salt []byte, mode cryptDirection) (func(io.Reader) ([]byte, error), error) {
-	cipher, err := buildBlowfish2Cipher(key, salt)
+	bfCipher, err := buildBlowfish2Cipher(key, salt)
 	if err != nil {
 		return nil, err
 	}
-	blocksize := blowfish.BlockSize
-	decrypt := func(reader io.Reader) (decryped []byte, err error) {
-		block0 := make([]byte, blocksize)
-		n, err := io.ReadFull(reader, block0)
+	decrypt := func(reader io.Reader) ([]byte, error) {
+		iv := make([]byte, bfCipher.BlockSize())
+		n, err := io.ReadFull(reader, iv)
 		if err != nil && n == 0 {
-			return
+			return nil, fmt.Errorf("unable to read iv: %w", err)
 		}
-		block1 := make([]byte, blocksize)
-		n, err = io.ReadFull(reader, block1)
-		if err != nil && n == 0 {
-			return
-		}
-		err = nil
 
-		for {
-			buf := append([]byte(nil), block0...)
-			buf, _ = convertEndian(buf)
-			cipher.Encrypt(buf, buf)
-			buf, _ = convertEndian(buf)
-			xorBytes(buf, buf, block1, blocksize)
-			decryped = append(decryped, buf[:n]...)
-
-			switch mode {
-			case bfEncrypt:
-				block0 = buf[:n]
-			case bfDecrypt:
-				block0 = block1
-			}
-			block1 = make([]byte, blocksize)
-			n, _ = io.ReadFull(reader, block1)
-			if n == 0 {
-				return
-			}
+		var stream cipher.Stream
+		if mode == bfEncrypt {
+			stream = cipher.NewCFBEncrypter(bfCipher, iv)
+		} else {
+			stream = cipher.NewCFBDecrypter(bfCipher, iv)
 		}
+		sReader := &cipher.StreamReader{S: stream, R: reader}
+		return io.ReadAll(sReader) //nolint:wrapcheck // caller wraps
 	}
 	return decrypt, nil
 }
 
-func buildBlowfish2Cipher(key, salt []byte) (cipher *blowfish.Cipher, err error) {
+func buildBlowfish2Cipher(key, salt []byte) (cipher.Block, error) {
 	pw := append([]byte(nil), key...)
 	for i := 1; i <= 1000; i++ {
 		hash := sha256.Sum256(append(pw, salt...))
 		pw = []byte(fmt.Sprintf("%x", hash))
 	}
 	hash := sha256.Sum256(append(pw, salt...))
-	cipher, err = blowfish.NewCipher(hash[:])
-	return
+	bfCipher, err := blowfish.NewCipher(hash[:])
+	return &VimBlowfish{bfCipher}, err //nolint:wrapcheck // can't happen
```

Comment:
That really seems to defeat the point of a linter, a linter aims to make code easier to maintain. It can't happen, but it could if someone changed the hash algorithm passed into NewCipher. By hiding the error if something did change, it becomes very hard for someone to debug.

I kind of don't care, but why are you applying a linter if you don't want quality code?